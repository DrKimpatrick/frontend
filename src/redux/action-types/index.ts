// This is where we will add action types