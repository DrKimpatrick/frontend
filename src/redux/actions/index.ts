// This is where we will add our action creators
export {};
