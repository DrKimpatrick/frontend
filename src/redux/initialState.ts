import users from 'redux/initialStates/users';

const initialState = {
  users
};

export default initialState;
