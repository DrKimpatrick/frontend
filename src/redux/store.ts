import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import apiMiddleware from 'middleware/apiMiddleware';
import rootReducer from './reducers/rootReducer';
import initialState from './initialState';

const middleware = [apiMiddleware];

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export type RootState = ReturnType<typeof rootReducer>;

export default store;
