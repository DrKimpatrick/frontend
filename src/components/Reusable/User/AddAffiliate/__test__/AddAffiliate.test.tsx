import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup, render } from '@testing-library/react';
import Component from '../AddAffiliate';
import { Store } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

let store: Store;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const initialState = {
  users: {
    addAffiliate: true,
    addAffiliateLoading: true,
    errors: {
      message: 'error message',
      errors: [{ username: 'username is required' }]
    }
  },
  messages: {
    message: 'uploaded'
  }
};
const props = {
  closeModal: jest.fn()
};
describe('AddAffiliate', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={store}>
        <Component {...props} />
      </Provider>,
      div
    );
  });

  it('should create snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Component {...props} />
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
