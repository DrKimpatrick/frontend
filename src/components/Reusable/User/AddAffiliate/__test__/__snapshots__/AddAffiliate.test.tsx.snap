// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAffiliate should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      role="presentation"
      style="position: fixed; z-index: 100; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="addItem"
        tabindex="-1"
      >
        <div
          class="flex relative h-auto heading"
        >
          <button
            class="back-arrow cursor-pointer"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </button>
          <h1
            class="font-bold text-base title"
          >
            Create affiliate user
          </h1>
        </div>
        <div
          class="addAffiliate"
        >
          <div
            class="inputError"
          >
            error message
          </div>
          <form
            autocomplete="off"
          >
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="username"
                placeholder="Username"
                type="text"
                value=""
              />
              <div
                class="inputError"
              >
                username is required
              </div>
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
              <div
                class="inputError"
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="paypalEmail"
                placeholder="Paypal email"
                type="email"
                value=""
              />
              <div
                class="inputError"
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="linkToPlatform"
                placeholder="Website (https://example.com)"
                type="url"
                value=""
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="password"
                placeholder="password"
                type="password"
                value=""
              />
              <div
                class="inputError"
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <textarea
                name="bio"
                placeholder="Bio"
              />
            </div>
            <div
              class="text-gray-texts mt-4 flex flex-col items-center justify-center"
            >
              <div
                class="uploadProfilePicture mt-4"
              >
                <input
                  type="file"
                />
                <button
                  type="button"
                >
                  Add profile picture
                </button>
                <div
                  class="inputError"
                />
              </div>
              <button
                class="mt-4 submitButton"
                disabled=""
                type="submit"
              >
                <div
                  class="css-1g85zfd"
                >
                  <div
                    class="css-krnedn"
                  />
                  <div
                    class="css-1oeuq11"
                  />
                  <div
                    class="css-krnedn"
                  />
                </div>
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
    <div
      aria-hidden="true"
    />
    <div
      role="presentation"
      style="position: fixed; z-index: 100; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="addItem"
        tabindex="-1"
      >
        <div
          class="flex relative h-auto heading"
        >
          <button
            class="back-arrow cursor-pointer"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </button>
          <h1
            class="font-bold text-base title"
          >
            Create affiliate user
          </h1>
        </div>
        <div
          class="addAffiliate"
        >
          <div
            class="inputError"
          >
            error message
          </div>
          <form
            autocomplete="off"
          >
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="username"
                placeholder="Username"
                type="text"
                value=""
              />
              <div
                class="inputError"
              >
                username is required
              </div>
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
              <div
                class="inputError"
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="paypalEmail"
                placeholder="Paypal email"
                type="email"
                value=""
              />
              <div
                class="inputError"
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="linkToPlatform"
                placeholder="Website (https://example.com)"
                type="url"
                value=""
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <input
                name="password"
                placeholder="password"
                type="password"
                value=""
              />
              <div
                class="inputError"
              />
            </div>
            <div
              class="text-gray-texts mt-4"
            >
              <textarea
                name="bio"
                placeholder="Bio"
              />
            </div>
            <div
              class="text-gray-texts mt-4 flex flex-col items-center justify-center"
            >
              <div
                class="uploadProfilePicture mt-4"
              >
                <input
                  type="file"
                />
                <button
                  type="button"
                >
                  Add profile picture
                </button>
                <div
                  class="inputError"
                />
              </div>
              <button
                class="mt-4 submitButton"
                disabled=""
                type="submit"
              >
                <div
                  class="css-1g85zfd"
                >
                  <div
                    class="css-krnedn"
                  />
                  <div
                    class="css-1oeuq11"
                  />
                  <div
                    class="css-krnedn"
                  />
                </div>
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
